annubes/task.py

```diff
@@ -330,25 +332,28 @@ def _check_int_positive(self, name: str, value: Any, strict: bool) -> None:  # n
             msg = f"`{name}` must be greater than or equal to 0."
             raise ValueError(msg)
 
-    def _check_bool(self, name: str, value: Any) -> None:  # noqa: ANN401
-        if not isinstance(value, bool):
-            msg = f"`{name}` must be a boolean."
-            raise TypeError(msg)
+    def _check_range(self, name: str, value: Any) -> None:  # noqa: ANN401
+        if isinstance(value, tuple):
+            for v in value:
+                self._check_int_positive(name, v, strict=False)
+        else:
+            self._check_int_positive(name, value, strict=False)
```

Comment:
Like this, if a list is passed, then the error message will tell you "must be an integer", while a tuple is also allowed. 
How about this?

Also, there should probably be an length check as well?

```suggestion
    def _check_range(self, name: str, value: Any) -> None:  # noqa: ANN401
        try:
            self._check_int_positive(name, value, strict=False)
        except (TypeError, ValueError):
            try:
                value = tuple(value)
                assert len(value) != 2  # noqa: S101
            except (TypeError, AssertionError) as e:
                msg = f"`{name}` must be an int or tuple of length 2."
                raise TypeError(msg) from e
            for v in value:
                self._check_int_positive("range value", v, strict=False)
```

Reply:
It is always better to avoid try: ... except (specific exceptions) if not strictly needed (you may not have considered all the possible exceptions). I edited it, in a way that `_check_range` is used everywhere (`iti`, `fix_time`, `ntrials`), it checks on the len if you have a tuple, and always print an error message with all the needed info (about integer + tuple of 2 integers).

Reply:
> It is always better to avoid try: ... except [specific exceptions] if not strictly needed (e.g., you may not have considered all the possible exceptions, you may not want to consider all the possible exceptions, it adds a logic block).

I often read that duck-typing is preferable over `isinstance` in most cases (although I guess the [formal stance](https://realpython.com/python-lbyl-vs-eafp/#the-pythonic-way-to-go-lbyl-or-eafp) is that both are equal). One of the reasons stated is that it's often less likely to lead to unexpected behavior. You can always make the exception even more specific by including the error message if you want to be even more sure not to catch the wrong exception (there is actually a ruff rule to enforce that, but I turned it off as it seemed overly strict to me).

As it is implemented now, if a float (with 0 behind the decimal) or list is passed, it will fail. Why do we not want to allow that? I think as long as it is obvious what was intended (like passing `[1, 10]` to a range rather than `(1, 10)`
Thinking about it now, the `_check_int_positive` method should probably allow floats that are ints anyway (see my comment)

Reply:
Duck-typing does not imply overusing try... except statements. Allowing all the possible cases is in my opinion much more prone to dev errors and requires more code in the code-base. If a tuple is required, it is fine to throw an error if the user passes a list. MinMaxScaler, for naming an example of a well established package, also does that. It simplifies the code a lot (imagine handling all the possible cases for many many parameters). If a tuple with integers is required, I am also in favor of throwing an error if floats are used. It means that the user is not getting what the parameter is, if floats are used.